node/src/components/consensus/highway_core/highway.rs

```diff
@@ -260,8 +262,10 @@ impl<C: Context> Highway<C> {
         rng: &mut dyn CryptoRngCore,
     ) -> Vec<Effect<C>> {
         let instance_id = self.instance_id.clone();
+        let timestamp = block_context.timestamp();
```

Comment:
@afck When proposing a block, we should just use its timestamp, which will be a little older than `Timestamp::now()`...?

Reply:
Yes, see comment above. What happens is basically this:
* We realize it's our turn to propose a block in millisecond 64, so we set a timer.
* The timer for timestamp 64 fires, and we request deploys for the new block from the deploy buffer (with 64 in the block context).
* The deploy buffer responds and we finally end up here, and can propose the new block. But we still have to use timestamp 64.